### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by brettcannon**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.linting.enabled": false
```
Comment (bityob) [inline .vscode/settings.json]: .vscode folder should be ignored (add to .gitignore)

Comment (brettcannon) [reply .vscode/settings.json]: Yep, but I also figured you wouldn't merge this PR anyway due to the review comments, so I didn't worry about it. 😄

File: azure-kusto-ingest/azure/kusto/ingest/kusto_ingest_client.py [hunk 2]
```diff
@@ -154,22 +160,26 @@ def ingest_from_multiple_blobs(self, blobs, delete_sources_on_success, ingestion
...
 
     def _refresh_containers_if_needed(self):
-        if (self._last_time_refreshed_containers > datetime.utcnow() - timedelta(hours=2)
+        if (self._last_time_refreshed_containers > datetime.datetime.utcnow() - datetime.timedelta(hours=2)
                 or not self._temp_storage_objects):
             self._last_time_refreshed_containers = datetime.utcnow()
             self._temp_storage_objects = self._get_temp_storage_objects()
 
     def _get_temp_storage_objects(self):
+        # review: where does self._kusto_client get set?
```
Comment (toshetah) [inline azure-kusto-ingest/azure/kusto/ingest/kusto_ingest_client.py]: in the __init__ method

Comment (brettcannon) [reply azure-kusto-ingest/azure/kusto/ingest/kusto_ingest_client.py]: Yeah, sorry for not noticing that.

File: azure-kusto-ingest/azure/kusto/ingest/kusto_ingest_client.py [hunk 3]
```diff
@@ -181,9 +191,9 @@ def _get_queues(self):
         return queues
 
     def _refresh_token_if_needed(self):
-        if (self._last_time_refreshed_token > datetime.utcnow() - timedelta(hours=2)
+        if (self._last_time_refreshed_token > datetime.datetime.utcnow() - datetime.timedelta(hours=2)
```
Comment (toshetah) [inline azure-kusto-ingest/azure/kusto/ingest/kusto_ingest_client.py]: I import datetime, is there any advantage taking this approach?

File: azure-kusto-data/azure_bdist_wheel.py [hunk 4]
```diff
@@ -7,13 +7,19 @@
...
+
+# review: class names should be CapWords (PEP 8)
 class azure_bdist_wheel(bdist_wheel):
+    # review: docstrings should start with a one-sentence summary (PEP 8, 257).
     """The purpose of this class is to build wheel a little differently than the sdist,
     without requiring to build the wheel from the sdist (i.e. you can build the wheel
     directly from source).
     """
 
+    # review: what's this for? What is this providing that the docstring doesn't?
```
Comment (toshetah) [inline azure-kusto-data/azure_bdist_wheel.py]: Actually this code was written by @lmazuel , maybe he can enlight us.

File: azure-kusto-data/azure/kusto/data/version.py [hunk 5]
```diff
@@ -1,2 +1,4 @@
 """ Version file. """
+# review: why do you have a separate file for the version? Why can't you just
+# inline this in __init__.py?
```
Comment (toshetah) [inline azure-kusto-data/azure/kusto/data/version.py]: I need it at kusto_client

File: azure-kusto-data/azure/kusto/data/kusto_client.py [hunk 6]
```diff
@@ -197,6 +223,11 @@ class KustoClient(object):
     >>>    print(row["ColumnName"])    """
 
     def __init__(self, kusto_cluster, client_id=None, client_secret=None, username=None, password=None, version='v1', authority=None):
+        # review: why are you suddenly  documenting the individual parameters?
+        # And don't document the types unless you really mean it. And if you
+        # do then make them type hints and use mypy to help keep you honest.
+        # And does this follow any known docstring formatting style in the
+        # Python community (e.g. Sphinx, Google, etc.)?
```
Comment (toshetah) [inline azure-kusto-data/azure/kusto/data/kusto_client.py]: I used to do it, but it is not compatible with python 2.7

File: azure-kusto-data/azure/kusto/data/version.py (hunk 5 above)
Comment (brettcannon) [reply azure-kusto-data/azure/kusto/data/version.py]: I would actually put all files in the `data/` folder into a single `data.py` and drop the subpackage all together.

File: azure-kusto-data/azure/kusto/data/kusto_client.py (hunk 6 above)
Comment (brettcannon) [reply azure-kusto-data/azure/kusto/data/kusto_client.py]: @toshetah what's not compatible with Python 2.7? You can provide type hints for Python 2.7 if that's what you mean using a comment.

File: azure-kusto-data/azure_bdist_wheel.py [hunk 7]
```diff
@@ -7,13 +7,19 @@
 from distutils import log as logger
 import os.path
 
+# review: the 'wheel' package has no officially public API, so be aware that this could break at any time.
```
Comment (lmazuel) [inline azure-kusto-data/azure_bdist_wheel.py]: @brettcannon I know, but that's the way we support setuptools and wheel with namespace packages across all SDK. They all have this file.

File: azure-kusto-data/azure_bdist_wheel.py [hunk 8]
```diff
@@ -7,13 +7,19 @@
...
 from wheel.bdist_wheel import bdist_wheel
+# review: two newlines between top-level definitions (PEP 8).
```
Comment (lmazuel) [inline azure-kusto-data/azure_bdist_wheel.py]: ok

File: azure-kusto-data/azure_bdist_wheel.py [hunk 9]
```diff
@@ -7,13 +7,19 @@
...
+
+# review: class names should be CapWords (PEP 8)
```
Comment (lmazuel) [inline azure-kusto-data/azure_bdist_wheel.py]: hum, I tried to follow the setuptools convention, so naming the azure version from "azure_bdist_wheel" from "bdist_wheel". That's pure packaging, and no one except the setup.py will use that anyway.

File: azure-kusto-data/azure_bdist_wheel.py [hunk 10]
```diff
@@ -7,13 +7,19 @@
...
 class azure_bdist_wheel(bdist_wheel):
+    # review: docstrings should start with a one-sentence summary (PEP 8, 257).
```
Comment (lmazuel) [inline azure-kusto-data/azure_bdist_wheel.py]: ok

**Pull request closed without merging**